### ISSUE TITLE
LayoutLMv2: Retrieve the text

### ISSUE THREAD

**Issue by alexdess** [NielsRogge/Transformers-Tutorials#59]
When inferring, what is the easiest way to recover the detected text? Indeed I recover the label and the boxes but not the text itself.

```
encoded_inputs = processor2(image_test, padding="max_length", truncation=True, return_tensors="pt", return_offsets_mapping=True)
offset_mapping = encoded_inputs.pop('offset_mapping')
print(encoded_inputs.keys())
```

> dict_keys(['input_ids', 'token_type_ids', 'attention_mask', 'bbox', 'image'])

If this is not possible, I will use tesseract upstream and use the processor without OCR.

Thanks again for your tutorials!

**Comment by NielsRogge** [comment]
You can decide the text using the tokenizer:

```
from transformers import LayoutLMv2Processor

processor.batch_decode(encoded_inputs.input_ids)
```

**Comment by alexdess** [comment]
Hi,

Thanks for your answer, unfortunately the LayoutLMv2Processor does not have a batch_decode function.

![image](https://user-images.githubusercontent.com/8177780/146894638-d4b0cced-9480-44b7-b566-c82c51a314a5.png)

Am I doing something wrong?

**Comment by alexdess** [comment]
I found the solution, it is not necessary to pass by the processor but directly by the tokenizer, the following code works:
`tokenizer2.batch_decode(encoded_inputs.input_ids)`

In general I encountered different problems to pass my layoutlmv2 pipeline to layoutxlm that I was able to solve by creating the processor with the tokenizer and the feature extractor. In particular those of the #58 issue
`feature_extractor2 = LayoutLMv2FeatureExtractor(apply_ocr=True)
tokenizer2 = LayoutXLMTokenizer.from_pretrained("microsoft/layoutxlm-base", model_max_length=512)
processor2 = LayoutXLMProcessor(feature_extractor2, tokenizer2)`

Thank you again for these examples that allowed me to complete my Master's work.
